semantics/src/main/scala/aqua/semantics/rules/types/TypesInterpreter.scala

```diff
@@ -58,6 +58,12 @@ class TypesInterpreter[S[_], X](using
         report.error(token, s"Unresolved type").as(None)
     }
 
+  def resolveNamedType(token: TypeToken[S]): State[X, Option[Type]] =
```

Comment:
I believe result type is better to be `NamedType` instead of `Type`, or even `AbilityType | StrcutType`, to avoid pattern matching on call site